src/platform/unix/mod.rs

```diff
@@ -500,6 +501,8 @@ impl OsIpcReceiverSet {
                         }
                         Err(err) if err.channel_is_closed() => {
                             self.pollfds.remove(&evt_token).unwrap();
+                            let io = EventedFd(&poll_entry.fd);
+                            self.poll.deregister(&io).unwrap();
```

Comment:
After I thought about it and looked at the code more, I realized the real problem is that we don't deregister the descriptor when we hit an error here. Theoretically just deregistering the descriptor here should resolve the issue.